web/controller.go

```diff
@@ -34,3 +37,24 @@ type Endpoint struct {
 	Method string
 	Path   string
 }
+
+type CredentialsExtractor interface {
+	Extract(*http.Request) (string, error)
+}
+
+type FuncCredentialsExtractor func(*http.Request) (string, error)
```

Comment:
CredentialsExtractorFunc looks better, and dont forget the comments on exported types